src/components/planet-description/index.tsx

```diff
@@ -0,0 +1,106 @@
+import { ReactNode, Suspense } from 'react';
+import {
+  useLocation,
+  useNavigate,
+  Params,
+  defer,
+  useLoaderData,
+  Await,
+} from 'react-router-dom';
+
+import Spinner from '../spinner';
+import ErrorMessage from '../error-message';
+
+import SwapiService from '../../services/swapi-service';
+
+import { IPlanetDescription } from '../../types/planet';
+import { DeferredData } from '@remix-run/router/dist/utils';
+
+import closeImg from '../../assets/image/close.png';
+
+import './planet-description.scss';
+
+export const planetDescriptionLoader = async ({
+  params,
+}: {
+  params: Params;
+}): Promise<DeferredData | undefined> => {
+  const { name } = params;
+  if (!name) {
+    return;
+  }
+  const res = SwapiService.getPlanetByName(name);
+  return defer({ res });
+};
+
+const PlanetDescription = (): ReactNode => {
```

Comment:
you don't need to describe that a function returns a React node. 
Better follow this syntax: `const PlanetDescription: React.FC<Props> = () => {}`